### PR TITLE
Toolbar: move all subcomponents under the same folder

### PR DESCRIPTION
**Pull request by ciampo**
## What?

Unify under the same parent `toolbar` folder all related subcomponents.

## Why?

This PR is in preparation to two refactors:

- unifying Storybook examples while moving away from knobs
- rewriting the components in TypeScript

Moving all of the subcomponents under the same folder will reduce overhead during those two refactors, while keeping all these changes in a separate PR will also greatly reduce diff noise.

## How?

- moved each `Toolbar*` subcomponent in its own folder, under the `toolbar` folder
- moved all Storybook examples under the `stories` folder
- moves all tests under the `tests` folder
- updated import & export paths across the project
- removed out-of-date paths from `tsconfig.json`

## Testing Instructions

- Make sure the project builds correctly and all tests pass (both web and native versions)
- Make sure Storybook examples work exactly like on `trunk`

### CODE REVIEW

File: packages/components/src/index.native.js [hunk 1]
```diff
@@ -15,12 +15,14 @@ export { default as Dashicon } from './dashicon';
...
-export { default as ToolbarItem } from './toolbar-item';
-export { default as ToolbarDropdownMenu } from './toolbar-dropdown-menu';
+export {
+	Toolbar,
+	ToolbarButton,
+	ToolbarContext as __experimentalToolbarContext,
+	ToolbarDropdownMenu,
+	ToolbarGroup,
+	ToolbarItem,
+} from './toolbar';
```
Comment (ciampo) [inline packages/components/src/index.native.js:25]: @geriux quick ping to make sure that the native side still builds and works like before

File: packages/components/src/toolbar/index.js [hunk 2]
```diff
@@ -1,52 +1,6 @@
...
-	);
-}
-
-export default forwardRef( Toolbar );
+export { default as Toolbar } from './toolbar';
+export { default as ToolbarButton } from './toolbar-button';
+export { default as ToolbarContext } from './toolbar-context';
+export { default as ToolbarDropdownMenu } from './toolbar-dropdown-menu';
+export { default as ToolbarGroup } from './toolbar-group';
+export { default as ToolbarItem } from './toolbar-item';
```
Comment (ciampo) [inline packages/components/src/toolbar/index.js:6]: For ease of review: the contents of this file have simply been moved to `packages/components/src/toolbar/toolbar/index.js` (the only code changes would be in the import paths).

A new `index.js` file was then added to re-export all components out of the `toolbar` folder.

Reviewing this PR commit-by-commit may be simpler.

**Review by mirka — APPROVED** [review]
Good call to do this prep work separately 👍 

Looks good, should be good to go once the `docs/manifest.json` is updated.

File: packages/components/src/index.native.js (hunk 1 above)
Comment (geriux) [reply packages/components/src/index.native.js:25]: I'll take a look 👍

Comment (geriux) [reply packages/components/src/index.native.js:25]: All working well 👍

**Pull request merged**